src/pages/previous.js

```diff
@@ -0,0 +1,152 @@
+import NavBar from "../components/navBar";
+import Chevron from "../assets/chevron_down.svg";
+import { GET } from "../utils/fetch";
+import { useEffect, useState } from "react";
+
+async function getQuestions() {
+  const weekly_questions = await GET("weekly_questions");
+  const questions = [];
+  for (const value in weekly_questions) {
+    questions.push({
+      id: weekly_questions[value].question_id,
+      body: weekly_questions[value].question.body,
+      time: new Date(weekly_questions[value].week),
+    });
+  }
+  return questions;
+}
+
+const options = {
+  month: "long",
+};
+
+function Previous() {
+  const [questions, setQuestions] = useState([]);
+  const [answers, setAnswers] = useState({});
+  const [answersVisibility, setAnswersVisibility] = useState({});
+
+  useEffect(() => {
+    getQuestions().then((response) => {
+      setQuestions(response);
+    });
+  }, []);
+
+  const viewAnswers = (questionId) => {
+    setAnswersVisibility((prevVisibility) => ({
+      ...prevVisibility,
+      [questionId]: !prevVisibility[questionId],
+    }));
+
+    getAll(questionId);
+  };
+
+  async function getAll(questionId) {
```

Comment:
```suggestion
  async function getAllAnswers(questionId) {
```

Is this what it is doing?

Reply:
Not only answers but as well the user that made the answer

Reply:
Well, but if that is part of the answer object, it's still answers, right?

Reply:
Well yes, in the the answers it only comes with the user_id so the email of the user I had to do another request

Reply:
But that can change in the API, right? There is a balance between what the API will provide by default, so it isn't too much stuff, but is still enough so that the FE doesn't have to make many requests.

A common solution (just so you know, not needed here), used by both JSON:API and GraphQL is to allow the FE to decide what they want to receive on a per request basis.

In this case, I think it makes sense to have the email of the user as well. Unless you already have it, otherwise you then have to fetch all the users for the ids you've just gotten.

Reply:
Yes yes, that's what I changed, yea much easier how it is now, better to take a bit more time on the BE to give what the FE needs, going forward will always have that in mind!